### PR TITLE
fix: restore performance of macOS window resizing

### PR DESCRIPTION
**Pull request by MarshallOfSound**
As in title, technically an upstream of a discord downstream patch --> https://github.com/discord/electron/commit/a712b6607e77c55b77aa8884e191a00001a697c5

Fixes an ongoing perf issue with window resizing on macOS by reverting an android-only fix in Chromium

Notes: Fixed resizing performance issue on macOS

### CODE REVIEW

**Review by VerteDinde — APPROVED** [review]

**Comment by nornagon** [comment]
has this been fixed upstream too?

**Comment by MarshallOfSound** [comment]
> has this been fixed upstream too?

No, it doesn't impact upstream, but I'm figuring out if I can make this change behind an desktop vs mobile check or something and upstream that.  For now this is a lightweight patch that fixes a rather annoying bug on our side with low risk (because it's reverting an android only fix)

**Comment by zcbenz** [comment]
Do you have a screencast that shows the difference? It should be possible to have Chromium revert the change for at least desktop if you can show them the downgraded performance.

**Comment by bpasero** [comment]
One thing I notice with Electron apps is how slow the content responds to resize events when quickly done rapidly:

Recording 2023-11-22 at 07 19 19

Curious if this change would have an impact 🤔

**Comment by MarshallOfSound** [comment]
> Curious if this change would have an impact 🤔

This is exactly what this fixes

**Comment by MarshallOfSound** [comment]
> Do you have a screencast that shows the difference? It should be possible to have Chromium revert the change for at least desktop if you can show them the downgraded performance.

I'll include a screencast when I bug this upstream 👍

**Comment by deepak1556** [comment]
> No, it doesn't impact upstream,

@MarshallOfSound curious why this issue does not impact desktop upstream ? Is there anything unique to the code path in question that has behavior differences between electron and chrome ?

**Comment by MarshallOfSound** [comment]
> curious why this issue does not impact desktop upstream ? Is there anything unique to the code path in question that has behavior differences between electron and chrome ?

I was interested in this too but couldn't figure it out.   There's definitely something different here (which would be good to figure out separately).  Will dig in deeper tomorrow.

File: patches/chromium/fix_restore_original_resize_performance_on_macos.patch [hunk 1]
```diff
@@ -0,0 +1,25 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Samuel Attard <marshallofsound@electronjs.org>
+Date: Tue, 21 Nov 2023 15:03:21 -0800
+Subject: fix: restore original resize performance on macOS
+
+Reverts https://chromium-review.googlesource.com/c/chromium/src/+/3118293 which
+fixed a android only regression but in the process regressed electron window
+resize perf wildly on macOS.
```
Comment (nornagon) [inline patches/chromium/fix_restore_original_resize_performance_on_macos.patch:8]: Per our [patch policy](https://github.com/electron/electron/blob/main/docs/development/patches.md#patch-justification), all patches are required to include in their description either (a) conditions for the patch's removal, or (b) a reason why the patch can't be removed. Please update this description 🙇🏻

**Comment by bpasero** [comment]
I just played around with the newly released Electron `27.1.2` and unfortunately do not see a difference:

Recording 2023-11-23 at 09 10 01


Filed https://github.com/electron/electron/issues/40603, because I cannot reproduce in web only.

**Comment by dtzxporter** [comment]
I would also like to mention that the issue we commented on in #36280 is also not resolved by this PR using 27.1.2: 
image

**Comment by cyhnkckali** [comment]
Same error persists in Windows

**Comment by nkallen** [comment]
The problem persists and in fact I would say it's gotten significantly worse on Mac as of Electron 29 and on Windows as of Electron 28. There is a perhaps related chromium bug that is not yet fixed:

https://issues.chromium.org/issues/376296999

**Pull request merged**